sycl/source/detail/event_impl.cpp

```diff
@@ -136,6 +136,7 @@ event_impl::event_impl(const QueueImplPtr &Queue) : MQueue{Queue} {
       if (!MHostProfilingInfo)
         throw runtime_error("Out of host memory", PI_OUT_OF_HOST_MEMORY);
     }
+    MIsProfilingEnabled = true;
```

Comment:
This should also be true if `MIsProfilingEnabled` on `Queue` is true. That is, you could move it outside the conditional and make it
```
MIsProfilingEnabled = Queue->is_host() || Queue->is_profiling_enabled();
```
where `is_profiling_enabled` would return `MIsProfilingEnabled` in `queue_impl`. Alternatively you could make `MIsProfilingEnabled` in `queue_impl` public, just like `MDiscardEvents` is.

Reply:
I got it. Thanks for your comment. Working on it.